### PR TITLE
add hydra support

### PR DESCRIPTION
**Pull request by matthdsm**
Preliminary support for Ory Hydra,
if possible, I'd like some pointers on how to pass args to the backend creation function. 
As I mentioned in #1, Hydra services have no set host, so this needs to be configurable.

Please let me know what needs to change to make this usable.

Thanks
M

### CODE REVIEW

File: loginpass/hydra.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                }
+            else:
+                params = {k:data[k] for k in kwargs.get('param_keys')}
+
+            return UserInfo(params)
+
+    return Hydra
+
+
+Hydra = create_hydra_backend('hydra', 'hostname')
```
Comment (lepture) [inline loginpass/hydra.py]: newline at eof

Comment (lepture) [reply loginpass/hydra.py]: This `Hydra = ` line can be removed since it is not really usable.

**Comment by lepture** [comment]
Do not change anything in `django_example/website/settings.py` since it is not related. And it is better to add a test case for it.

File: loginpass/hydra.py (hunk 1 above)
Comment (matthdsm) [reply loginpass/hydra.py]: Hi @lepture 
Thank for the comments. Don't I need the `Hydra =` line so I can export the backend to `__init__.py`? 
Do you have any suggestions on how to fix the variable host issue?
Thanks again.
M

Comment (lepture) [reply loginpass/hydra.py]: You can just export `create_hydra_backend` to `__init__.py`, that is ok.

**Comment by matthdsm** [comment]
Thanks already for the help.
I've added some barebones test, but I still need to find a public hydra based test deployment, or anything that uses similar endpoints, so I can add test data and the "custom" hostname

M

**Comment by matthdsm** [comment]
Ready to merge IMO.

Cheers
M

**Comment by lepture** [comment]
Thanks

**Pull request merged**